tests/test_osut.py

```diff
@@ -649,6 +649,41 @@ def test05_construction_generation(self):
         self.assertEqual(o.status(), 0)
         del model
 
+        # Invalid Uo (here, skylights and windows inherit default Uo values)
+        specs = dict(type="skylight", uo=None)
+        model = openstudio.model.Model()
+        c = osut.genConstruction(model, specs)
+        self.assertEqual(o.status(), 0)
+        self.assertFalse(o.logs())
+        self.assertTrue(c)
+        self.assertTrue(isinstance(c, openstudio.model.Construction))
+        self.assertEqual(c.nameString(), "OSut.CON.skylight")
+        self.assertTrue(c.layers())
+        self.assertEqual(len(c.layers()), 1)
+        self.assertEqual(c.layers()[0].nameString(), "OSut.skylight.U3.5.SHGC45")
+        r = osut.rsi(c)
+        self.assertAlmostEqual(r, 1/osut.uo()["skylight"], places=3)
```

Comment:
In the case of fenestrated assemblies, an invalid Uo request is ignored and the glazing assembly inherits a default OSut Uo (here 3.5 W/m2.K for a skylight).